### PR TITLE
Docs/motion translate

### PR DESCRIPTION
**Pull request by 16stelter**
## Proposed changes
Updates and translates the lowlevel debugging docs and aptly renamed them.

## Necessary checks
- [ ] Update package version
- [x] Run `catkin build`
- [x] Write documentation
- [ ] Create issues for future work
- [x] Test on your machine
- [ ] Test on the robot
- [ ] Put the PR on our Project board

### CODE REVIEW

File: bitbots_docs/docs/manual/tutorials/lowlevel.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+If you have another good idea that can be an alternatives, Robotis is asking for advice via a github issue:  https://github.com/ROBOTIS-GIT/DynamixelSDK/issues
+
+
+Dynamixel SDK
+~~~~~~~~~~~~~
+
+The Dynamixel SDK implements the Dynamixel protocol.
+It provides methods to send instructions and to read status packets in different programming languages.
+We use a fork of Robotis' Dynamixel SDK because Robotis did not implement the sync read on multiple registers.
```
Comment (johagge) [inline bitbots_docs/docs/manual/tutorials/lowlevel.rst:106]: link the fork?

File: bitbots_docs/docs/manual/tutorials/lowlevel.rst [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+After all these steps we're finally at the ROS message level.
+There are two message types handled here: The joint state message gives the current positions of the motors, while the joint goals message can specify target positions for motors. The hardware interface also manages the IMU data and the values returned by the foot sensors.
+
+How do we use bitbots_ros_control?
+----------------------------------
+
+The package bitbots_ros_control provides the hardware interface for the dynamixel motors.
+
+The most important configuration file for this is the wolfgang.yaml. In this file you find multiple settings for defining, which values should be read from the motors (temperature, speed, force, ...), which sensors should be used (foot pressure sensors, IMU) and which settings should be set (control loop frequency, baud rate, CORE ports, auto torque, ...).
```
Comment (johagge) [inline bitbots_docs/docs/manual/tutorials/lowlevel.rst:134]: These sentences should probably be each on their own line.

File: bitbots_docs/docs/manual/tutorials/lowlevel.rst [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+The corresponding ROS node can be launched with `roslaunch bitbots_ros_control ros_control.launch`. This will execute the following steps:
+
+1. The motors are pinged in alphabetical order. This happens due to the way yaml files are read. This means the HeadPan motor (id 19) is read first, while the RShoulderRoll motor (id 3) is read last.
```
Comment (johagge) [inline bitbots_docs/docs/manual/tutorials/lowlevel.rst:138]:
```suggestion
1. The motors are pinged one after another in alphabetical order. This happens due to the way yaml files are read. This means the HeadPan motor (id 19) is read first, while the RShoulderRoll motor (id 3) is read last.
```

Comment (johagge) [reply bitbots_docs/docs/manual/tutorials/lowlevel.rst:138]: The sentences should probably each be on their own line.

File: bitbots_docs/docs/manual/tutorials/lowlevel.rst [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+2. The values from the config file are written into the RAM and ROM of the motors. These are values like speed or return delay time.
+3. The message "Hardware interface init finished" appears.
```
Comment (johagge) [inline bitbots_docs/docs/manual/tutorials/lowlevel.rst:140]:
```suggestion
3. The message "Hardware interface init finished" is printed to the terminal.
```
i assume?

File: bitbots_docs/docs/manual/tutorials/lowlevel.rst [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+4. The control loop starts, alternating between sync read and sync write.
+5. The controllers for ROS control are loaded.
+
+Help, I have a problem!
+-----------------------
+
+Error Opening Serial Port
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you encounter the message "Error opening serial port", no connection between the NUC and the CORE board could be established. Therefore your first instinct should be checking whether the cable is plugged in correctly. If this does not solve the problem, you can check whether the board can be found by using `lsusb` (look at the "leaf" entry). You can further investigate this by using `ls /dev/`. You should find the devices "/dev/ttyUSB0" through "/dev/ttyUSB3", one for each of the four busses. If the names are different, you may have to alter the wolfgang.yaml file or unplug the CORE board and plug it back in, in order to make it use the known names.
```
Comment (johagge) [inline bitbots_docs/docs/manual/tutorials/lowlevel.rst:150]: These sentences should probably each be their own line.

Comment (johagge) [reply bitbots_docs/docs/manual/tutorials/lowlevel.rst:150]:
```suggestion
If you encounter the message "Error opening serial port", no connection between the NUC and the CORE board could be established. Therefore your first instinct should be checking whether the cable is plugged in correctly. If this does not solve the problem, you can check whether the board can be found by using `lsusb` (look for the "leaf" entry). You can further investigate this by using `ls /dev/`. You should find the devices "/dev/ttyUSB0" through "/dev/ttyUSB3", one for each of the four busses. If the names are different, you may have to alter the wolfgang.yaml file or unplug the CORE board and plug it back in, in order to make it use the known names.
```

File: bitbots_docs/docs/manual/tutorials/lowlevel.rst [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+The ROS Control Framework is a part of ROS that is responsible for the motor and sensor control.
+There are controllers for ROS Control that provide the interface between ROS and low level software parts.
+These controllers are hardware agnostic because they are using interfaces to abstract from the hardware (e.g. motors).
+To control the motors, the Dynamixel Controller is used, which itself uses the Dynamixel Hardware Interface.
+
+ROS messages
+~~~~~~~~~~~~
+
+After all these steps we're finally at the ROS message level.
+There are two message types handled here: The joint state message gives the current positions of the motors, while the joint goals message can specify target positions for motors. The hardware interface also manages the IMU data and the values returned by the foot sensors.
```
Comment (johagge) [inline bitbots_docs/docs/manual/tutorials/lowlevel.rst:127]: These sentences should probably be each on their own line.

File: bitbots_docs/docs/manual/tutorials/lowlevel.rst [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+Help, I have a problem!
+-----------------------
+
+Error Opening Serial Port
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you encounter the message "Error opening serial port", no connection between the NUC and the CORE board could be established. Therefore your first instinct should be checking whether the cable is plugged in correctly. If this does not solve the problem, you can check whether the board can be found by using `lsusb` (look at the "leaf" entry). You can further investigate this by using `ls /dev/`. You should find the devices "/dev/ttyUSB0" through "/dev/ttyUSB3", one for each of the four busses. If the names are different, you may have to alter the wolfgang.yaml file or unplug the CORE board and plug it back in, in order to make it use the known names.
+
+Motor problems
```
Comment (johagge) [inline bitbots_docs/docs/manual/tutorials/lowlevel.rst:152]: In this chapter the sentences should each have their own line

File: bitbots_docs/docs/manual/tutorials/lowlevel.rst [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+~~~~~~~~~~~~~~
+
+The first thing you should do if you have a motor problem ("no status from id", motors stuttering, ...) is checking whether all cables are plugged in correctly. Sometimes one of the cable sits loosely in its socket and may fall out entirely when the robot moves. To control whether all motors are reachable, this `software
```
Comment (johagge) [inline bitbots_docs/docs/manual/tutorials/lowlevel.rst:155]:
```suggestion
The first thing you should do if you have a motor problem ("no status from id", motors stuttering, ...) is checking whether all cables in the bus with the problem are plugged in correctly. Sometimes one of the cable sits loosely in its socket and may fall out entirely when the robot moves. To control whether all motors are reachable, this `software
```

**Review by johagge — CHANGES_REQUESTED** [review]

File: bitbots_docs/docs/manual/tutorials/lowlevel.rst (hunk 8 above)
Comment (16stelter) [reply bitbots_docs/docs/manual/tutorials/lowlevel.rst:155]: This edit should not be added, as even unrelated cables can cause issues in other motors. Of course it makes sense to check the affected cables first, but in general it makes sense to go through all cables, at least of that bus, to make sure.

Comment (johagge) [reply bitbots_docs/docs/manual/tutorials/lowlevel.rst:155]:
```suggestion
The first thing you should do if you have a motor problem ("no status from id", motors stuttering, ...) is checking whether all cables are plugged in correctly. Preferably starting with the bus with the motor that has a problem. Sometimes one of the cable sits loosely in its socket and may fall out entirely when the robot moves. To control whether all motors are reachable, this `software
```

File: bitbots_docs/docs/manual/tutorials/lowlevel.rst [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+For that, different hardware and software components are used; they are described in this article.
+Additionally, common problems and troubleshooting strategies are explained.
+
+The behavior of the low level packages is dominated by the control loop, i.e. a cycle that alternately reads and writes motor positions.
+In parallel to this cycle, the new positions are calculated based on the observed values.
+
+To react to problems as fast as possible, e.g. to overload errors in the motors, a faster control loop is desirable.
+Formerly, our loop ran with a frequency of 100Hz, therefore the reaction time to errors it at least 20ms because the error has to be read, processed and written, which takes at least two cycles.
+
+There are essentially three possibilities to accelerate the control loop:
```
[the remark this answers is not in the document]
Comment (16stelter) [reply bitbots_docs/docs/manual/tutorials/lowlevel.rst:17]: all of them ;)

[the remark this answers is not in the document]
Comment (johagge) [reply bitbots_docs/docs/manual/tutorials/lowlevel.rst:17]:
```suggestion
There are essentially three possibilities which we use to accelerate the control loop:
```

**Review by Flova — APPROVED** [review]

**Review by johagge — APPROVED** [review]

**Pull request merged**